server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -1031,19 +1032,25 @@ private static void updateReplicationStrategy(
      */
     private static void updateRemoteStoreSettings(Settings.Builder settingsBuilder, Settings clusterSettings) {
         if (isRemoteDataAttributePresent(clusterSettings)) {
+            String segmentDataRepository = clusterSettings.get(
+                Node.NODE_ATTRIBUTES.getKey() + RemoteStoreNodeAttribute.REMOTE_STORE_SEGMENT_DATA_REPOSITORY_NAME_ATTRIBUTE_KEY
+            );
             settingsBuilder.put(SETTING_REMOTE_STORE_ENABLED, true)
+                .put(SETTING_REMOTE_SEGMENT_STORE_DATA_REPOSITORY, segmentDataRepository)
                 .put(
-                    SETTING_REMOTE_SEGMENT_STORE_REPOSITORY,
-                    clusterSettings.get(
-                        Node.NODE_ATTRIBUTES.getKey() + RemoteStoreNodeAttribute.REMOTE_STORE_SEGMENT_REPOSITORY_NAME_ATTRIBUTE_KEY
-                    )
-                )
-                .put(
-                    SETTING_REMOTE_TRANSLOG_STORE_REPOSITORY,
+                    SETTING_REMOTE_TRANSLOG_STORE_DATA_REPOSITORY,
                     clusterSettings.get(
-                        Node.NODE_ATTRIBUTES.getKey() + RemoteStoreNodeAttribute.REMOTE_STORE_TRANSLOG_REPOSITORY_NAME_ATTRIBUTE_KEY
+                        Node.NODE_ATTRIBUTES.getKey() + RemoteStoreNodeAttribute.REMOTE_STORE_TRANSLOG_DATA_REPOSITORY_NAME_ATTRIBUTE_KEY
                     )
                 );
+            String segmentMetadataRepository = clusterSettings.get(
+                Node.NODE_ATTRIBUTES.getKey() + RemoteStoreNodeAttribute.REMOTE_STORE_SEGMENT_METADATA_REPOSITORY_NAME_ATTRIBUTE_KEY
+            );
+            if (segmentMetadataRepository != null) {
+                settingsBuilder.put(SETTING_REMOTE_SEGMENT_STORE_METADATA_REPOSITORY, segmentMetadataRepository);
+            } else {
+                settingsBuilder.put(SETTING_REMOTE_SEGMENT_STORE_METADATA_REPOSITORY, segmentDataRepository);
```

Comment:
is this intended behavior or just kept to maintain backward compatibility? for newer versions we would want users to explicitly specify the metadata repo as well, even if its the same as data repository.

Reply:
This is mostly to maintain backward compatibility but we may not want to enforce metadata repository if both are same.